src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -215,11 +217,19 @@ public static DeclarationModifiers ToDeclarationModifiers(this SyntaxTokenList m
                 result |= one;
             }
 
-            // the two keywords "protected" and "internal" together are treated as one modifier.
-            if ((result & DeclarationModifiers.AccessibilityMask) == (DeclarationModifiers.Protected | DeclarationModifiers.Internal))
+            switch (result & DeclarationModifiers.AccessibilityMask)
             {
-                result &= ~DeclarationModifiers.AccessibilityMask;
-                result |= DeclarationModifiers.ProtectedInternal;
+                case DeclarationModifiers.Protected | DeclarationModifiers.Internal:
+                    // the two keywords "protected" and "internal" together are treated as one modifier.
+                    result &= ~DeclarationModifiers.AccessibilityMask;
+                    result |= DeclarationModifiers.ProtectedInternal;
+                    break;
+
+                case DeclarationModifiers.Private | DeclarationModifiers.Protected:
```

Comment:
No, we don't want to give any particular meaning to that combination; we want to let the caller report it as an error.